cmd/daemon.go

```diff
@@ -49,11 +51,20 @@ var startDaemonCmd = &cobra.Command{
 			return fmt.Errorf("daemon must be run as root")
 		}
 
-		var err error
+		ctx, cancel := context.WithCancel(cmd.Context())
+		defer cancel()
 
 		log.Info().Str("version", rootCmd.Version).Msg("starting daemon")
 
-		server, err := server.NewServer(cmd.Context(), &server.ServeOpts{
+		// Initialize metrics reporting if ASR flag is enabled
+		if viper.GetBool("metrics.asr") {
```

Comment:
Take a look at how we reference config items at other places. We don't use `viper.Get...` anymore. You can directly reference the `config.Global` definition as it's always updated with the right values.